homeassistant/components/heos/media_player.py

```diff
@@ -125,12 +134,22 @@ async def async_added_to_hass(self):
                 SIGNAL_HEOS_UPDATED, self._heos_updated
             )
         )
+        # Register this player's entity_id so it can be resolved by the group manager
+        self.hass.data[HEOS_DOMAIN][DATA_ENTITY_ID_MAP][
+            self._player.player_id
+        ] = self.entity_id
+        self._group_manager.force_update_groups()
```

Comment:
We should call the async api from async context.

Reply:
True. Since I have replaces the method with a signal I am now using `hass.helpers.dispatcher.async_dispatcher_send` right there instead. So we can keep the async context.